### PR TITLE
[SM] Complete tasks for module 1-4

### PR DESCRIPTION
**Pull request by Malahovskys**

### CODE REVIEW

File: module-1/classification.js [hunk 1]
```diff
@@ -22,9 +22,9 @@ function grade(score) {
      * Store the grade in the gradeOfStudent variable.
      * Also take into consideration the documentation of the function!
      */
-    // PLACE YOUR CODE BETWEEN THIS...
 
-    // ...AND THIS COMMENT LINE!
+    gradeOfStudent = score < 0 || score > 100 ? 0 : Math.max(Math.min(Math.floor(score/10-4), 5), 1);
```
Comment (szikszail) [inline module-1/classification.js:26]: Nice solution, please ping me on Skype4Business

File: module-1/euclidean.js [hunk 2]
```diff
@@ -17,10 +17,28 @@ function euclidean(a, b) {
      * If you have the result, assign it to the variable, called gcd.
      * Also take into consideration the documentation of the function!
      */
-    // PLACE YOUR CODE BETWEEN THIS...
 
+    if(a <= 0 || b <= 0){
```
Comment (szikszail) [inline module-1/euclidean.js:21]: It is quite an interesting implementation of the algorithm. It is not optimal, actually quite expensive.
For example, your algorithm for `euclidean(683,997)` will compute 1 in 996 iterations, although the original algorithm does it in 6.

**Review by szikszail — CHANGES_REQUESTED** [review]
Please check my comments.

File: module-1/classification.js (hunk 1 above)
Comment (Malahovskys) [reply module-1/classification.js:26]: Pinged you (Siarhei Malakhouski)

File: module-1/euclidean.js (hunk 2 above)
Comment (Malahovskys) [reply module-1/euclidean.js:21]: Agree, unfortunately no time to rework :(

**Comment by szikszail** [comment]
Thank you for your solution!

**Pull request closed without merging**